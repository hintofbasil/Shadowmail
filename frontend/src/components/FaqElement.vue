<template>
  <div class="offset-by-two eight columns faq-question">
    <details>
      <summary>{{ element.summary }}</summary>
      <p>{{ element.content }}</p>
    </details>
  </div>
</template>

<script>
export default {
  props: {
    element: Object
  }
};
</script>

<style lang="scss" scoped>
@import "../colors";

.faq-question {
  margin-top: 2%;
  margin-bottom: 2%;

  summary {
    display: list-item;
    font-size: 1.1em;
  }
}
</style>
