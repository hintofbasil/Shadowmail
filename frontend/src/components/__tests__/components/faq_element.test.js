import React from 'react';
import renderer from 'react-test-renderer';

import FaqElement from '../../faq_element';

describe('render', () => {
  it('renders successfully', () => {
    const tree = renderer
      .create(
        <FaqElement
          summary="summary"
          content="content"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
