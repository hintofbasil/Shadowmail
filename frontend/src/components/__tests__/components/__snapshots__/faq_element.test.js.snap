// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders successfully 1`] = `
<div
  className="row faq-container"
>
  <div
    className="offset-by-two eight columns faq-question"
  >
    <details>
      <summary>
        summary
      </summary>
      <p>
        content
      </p>
    </details>
  </div>
</div>
`;
